server/rpc.go

```diff
@@ -50,6 +51,81 @@ type Broadcaster interface {
 	Sign([]byte) ([]byte, error)
 }
 
+type BroadcastSessionsManager struct {
+	broadcastSessions []*BroadcastSession
+	sessLock          *sync.Mutex
+}
+
+// func (bsm *BroadcastSessionsManager) selectFromList(node *core.LivepeerNode, pl core.PlaylistManager) *BroadcastSession {
+func (bsm *BroadcastSessionsManager) selectFromList() *BroadcastSession {
+	numOrchs := 3
+	// bsm.checkStatusOfList(numOrchs, node, pl)
+	bsm.checkStatusOfList(numOrchs)
+
+	bsm.sessLock.Lock()
+	defer bsm.sessLock.Unlock()
+
+	sess, sessions := bsm.broadcastSessions[0], bsm.broadcastSessions[1:]
+	bsm.broadcastSessions = sessions
+
+	return sess
+}
+
+func (bsm *BroadcastSessionsManager) removeFromList(session *BroadcastSession) {
+	bsm.sessLock.Lock()
+	defer bsm.sessLock.Unlock()
+	var sessions []*BroadcastSession
+	for i, s := range bsm.broadcastSessions {
+		if s == session {
+			sessions = append(bsm.broadcastSessions[:i], bsm.broadcastSessions[i+1:]...)
+			break
+		}
+	}
+	bsm.broadcastSessions = sessions
+}
+
+func (bsm *BroadcastSessionsManager) addToList(sess *BroadcastSession) {
+	bsm.sessLock.Lock()
+	bsm.broadcastSessions = append(bsm.broadcastSessions, sess)
+	bsm.sessLock.Unlock()
+}
+
+// func (bsm *BroadcastSessionsManager) checkStatusOfList(numOrchs int, node *core.LivepeerNode, pl core.PlaylistManager) {
+func (bsm *BroadcastSessionsManager) checkStatusOfList(numOrchs int) {
+	reqLengthOfList := numOrchs * 3
+
+	var newBroadcastSessions []*BroadcastSession
+
+	bsm.sessLock.Lock()
+	if len(bsm.broadcastSessions) < numOrchs {
+		bsm.sessLock.Unlock()
+		// newBroadcastSessions, err := selectOrchestrator(node, pl)
```

Comment:
Once we integrate with selectOrchestrator, this function will become more relevant and will be properly tested.